lib/moment.plugin.js

```diff
@@ -1,13 +1,11 @@
-import Vue from 'vue'
 import moment from 'moment'
 
 <%= options.locales.map(l => `import 'moment/locale/${l}'`).join('\n') %>
 
-if (!Vue.prototype.hasOwnProperty('$moment')) {
-  Object.defineProperty(Vue.prototype, '$moment', {
-    get() {
-      return moment
-    }
-  })
-}
+<% if(options.defaultLocale) { %>
+  moment.locale('<%= options.defaultLocale %>')
+<% } %>
 
+export default (ctx, inject) => {
+  inject('moment', moment)
```

Comment:
Yes. I'm aware of the nuxt reload problem. Because of shared Vue instance, any config change for moment module is not being applied unless stop/start nuxt again. Anyway, I think it worth to fix the root problem instead of making production builds slower. We have 3 options:

- For now, use `inject` to prevent that problem and accept extra load until Nuxt.js fix
- Use a conditional if for the template to use non-inject method for production only
- Just making users aware of such development inconsistency and keep it as is

Reply:
Because several issues stated that ppl want to use it with `ctx`, so also on server side.

Reply:
@manniL Honestly this was not the purpose of this module and moment can be easily `imported` whenever needed which is fully SSR safe too. BTW I agree with that. Let's keep using inject :)